File: common/src/main/java/dev/number6/db/model/converter/LocalDateTimeConverter.java
package dev.number6.db.model.converter;

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter;

import java.time.LocalDateTime;

public class LocalDateTimeConverter implements DynamoDBTypeConverter<String, LocalDateTime> {

    @Override
    public String convert(LocalDateTime object) {
        return object.toString();
    }

    @Override
    public LocalDateTime unconvert(String object) {
        return LocalDateTime.parse(object);
    }
}

File: common/src/main/kotlin/dev/number6/db/model/converter/LocalDateTimeConverter.kt
package dev.number6.db.model.converter

import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverter
import java.time.LocalDateTime

class LocalDateTimeConverter : DynamoDBTypeConverter<String, LocalDateTime> {
    override fun convert(dateTime: LocalDateTime): String {
        return dateTime.toString()
    }

    override fun unconvert(dateTime: String): LocalDateTime {
        return LocalDateTime.parse(dateTime)
    }
}